src/io/export_pdf.cc

```diff
@@ -290,10 +291,22 @@ void export_pdf(const std::shared_ptr<const Geometry>& geom, std::ostream& outpu
   // Note Y axis + is DOWN.  Drawings have to invert Y, but these translations account for that.
   cairo_translate(cr, tcX, tcY);  // Center page on geometry;
 
-  cairo_set_source_rgba(cr, 0., 0., 0., 1.0); // Set black line, opaque
-  cairo_set_line_width(cr, 1);  // 1 point width.
-  draw_geom(geom, cr);
-  cairo_stroke(cr);
+  const Color4f black = Color4f(0.0f, 0.0f, 0.0f);
+
+  if (options->fill) {
+    Color4f fillColor = OpenSCAD::getColor(options->fillColor, black);
+    cairo_set_source_rgba(cr, fillColor[0], fillColor[1], fillColor[2], fillColor[3]);
+    draw_geom(geom, cr);
+    cairo_fill(cr);
+  }
+
+  if (options->stroke) {
+    Color4f strokeColor = OpenSCAD::getColor(options->strokeColor, black);
+    cairo_set_source_rgba(cr, strokeColor[0], strokeColor[1], strokeColor[2], strokeColor[3]);
+    cairo_set_line_width(cr, mm_to_points(options->strokeWidth));
+    draw_geom(geom, cr);
+    cairo_stroke(cr);
```

Comment:
If I'm reading the docs correctly, it should be possible to use `cairo_stroke_preserve()` which preserves the path, so a following fill should work without the need to redraw the geometry.

Reply:
Good catch, updated in next revision.

Reply:
Ah, cool, there's also a fill-preserve, I did not check that. Even better this way :+1:.